### PR TITLE
Add iBox2 scan

### PR DESCRIPTION
**Pull request by Erriez**

### CODE REVIEW

**Comment by Erriez** [comment]
@eljek I've implemented iBox2 scan functionality by reverse engineering as I could not find any documentation on internet. Do you have time to test it in branch `add-device-scan`? See updated protocol in `README.md`.

Note: I have to update the `/tests/test_milight_ibox2.py`.

Thanks!

File: milight_ibox2/milight_ibox2_control.py [hunk 1]
```diff
@@ -214,17 +267,17 @@ def send_command(self, light_command):
             self.ibox_seq &= 0xFF
 
             self.socket_send(cmd)
-            data = self.socket_recv()
-            if data:
-                if len(data) != 8:
+            rx_data, (rx_addr, rx_port) = self.socket_recv()
```
Comment (eljek) [inline milight_ibox2/milight_ibox2_control.py]: This unpacking will fail if None is received

File: milight_ibox2/milight_ibox2_control.py [hunk 2]
```diff
@@ -135,18 +188,18 @@ def connect(self):
                 else:
                     print("TX retry %d..." % retry)
             self.socket_send(cmd_start_session)
-            data = self.socket_recv()
-            if data:
-                if len(data) != 22:
+            rx_data, (rx_addr, rx_port) = self.socket_recv()
```
Comment (eljek) [inline milight_ibox2/milight_ibox2_control.py]: also here unpacking might fail if None is received

File: milight_ibox2/milight_ibox2_control.py [hunk 3]
```diff
@@ -109,7 +116,49 @@ def print_bytearray(data, msg=""):
...
+        found_ibox_devices = []
+        data = bytearray([0x13, 0x00, 0x00, 0x00, 0x0a, 0x03, 0x9b, 0x7f, 0x11, 0xf0, 0xfe, 0x6b, 0x3b, 0xdd, 0xd4])
+
+        self.socket_open()
+        self.socket_send(data, broadcast=True)
+
+        while 1:
+            try:
+                # Wait for UDP response from iBox
+                (data, (ibox_addr, ibox_port)) = self.sock_server.recvfrom(100)
```
Comment (eljek) [inline milight_ibox2/milight_ibox2_control.py]: potential runtime error if none is received

**Review by eljek — COMMENTED** [review]
Hi!
I attempted to test this feature, however I only own a single ibox2 device, so I did not receive anything back by the scan. This probably also depends on invididual network configurations and so on. However I did find some minor issue, that was causing a runtime error when the receive operation returns none

**Comment by Erriez** [comment]
Thanks for your valuable review feedback and time! I'll fix this.

File: milight_ibox2/milight_ibox2_control.py (hunk 1 above)
Comment (Erriez) [reply milight_ibox2/milight_ibox2_control.py]: I agree, and I've pushed a fix to this branch.

File: milight_ibox2/milight_ibox2_control.py (hunk 2 above)
Comment (Erriez) [reply milight_ibox2/milight_ibox2_control.py]: I agree, and I've pushed a fix to this branch.

File: milight_ibox2/milight_ibox2_control.py (hunk 3 above)
Comment (Erriez) [reply milight_ibox2/milight_ibox2_control.py]: I agree, and I've pushed a fix to this branch. 

Note: I could not find a situation to return type `None`. In Python3.9 an empty `bytearray` is returned.

**Pull request merged**